Date: Sun, 10 Apr 2011 15:46:08 +0200
Description: do not use ressources for *.ico files
  use installed icons instead of pulling them from the ressource.
  This avoids a UnknownImageFormat error probably caused by mono bug.
Bugs-mono: https://bugzilla.novell.com/show_bug.cgi?id=599462
Author: Julian Taylor <jtaylor.debian@googlemail.com>

---
 KeePass/Forms/AboutForm.cs              |    4 ++--
 KeePass/Forms/CharPickerForm.cs         |    2 +-
 KeePass/Forms/ColumnsForm.cs            |    2 +-
 KeePass/Forms/DataEditorForm.cs         |    2 +-
 KeePass/Forms/DataViewerForm.cs         |    2 +-
 KeePass/Forms/DatabaseOperationsForm.cs |    2 +-
 KeePass/Forms/DatabaseSettingsForm.cs   |    2 +-
 KeePass/Forms/EcasTriggerForm.cs        |    2 +-
 KeePass/Forms/EcasTriggersForm.cs       |    2 +-
 KeePass/Forms/EditAutoTypeItemForm.cs   |    2 +-
 KeePass/Forms/EditStringForm.cs         |    2 +-
 KeePass/Forms/EntropyForm.cs            |    2 +-
 KeePass/Forms/EntryListForm.cs          |    2 +-
 KeePass/Forms/EntryReportForm.cs        |    2 +-
 KeePass/Forms/ExchangeDataForm.cs       |    2 +-
 KeePass/Forms/FileBrowserForm.cs        |    2 +-
 KeePass/Forms/GroupForm.cs              |    2 +-
 KeePass/Forms/HelpSourceForm.cs         |    2 +-
 KeePass/Forms/IOConnectionForm.cs       |    2 +-
 KeePass/Forms/IconPickerForm.cs         |    2 +-
 KeePass/Forms/ImportCsvForm.cs          |    2 +-
 KeePass/Forms/ImportMethodForm.cs       |    2 +-
 KeePass/Forms/InternalBrowserForm.cs    |    2 +-
 KeePass/Forms/KeyCreationForm.cs        |    2 +-
 KeePass/Forms/KeyPromptForm.cs          |    2 +-
 KeePass/Forms/LanguageForm.cs           |    2 +-
 KeePass/Forms/MainForm.cs               |    2 +-
 KeePass/Forms/MainForm_Functions.cs     |   12 ++++++------
 KeePass/Forms/OptionsForm.cs            |    2 +-
 KeePass/Forms/PluginsForm.cs            |    2 +-
 KeePass/Forms/PrintForm.cs              |    2 +-
 KeePass/Forms/PwEntryForm.cs            |    2 +-
 KeePass/Forms/PwGeneratorForm.cs        |    2 +-
 KeePass/Forms/SearchForm.cs             |    2 +-
 KeePass/Forms/SingleLineEditForm.cs     |    2 +-
 KeePass/Forms/StatusLoggerForm.cs       |    2 +-
 KeePass/Forms/TanWizardForm.cs          |    2 +-
 KeePass/Forms/UrlSchemeForm.cs          |    2 +-
 KeePass/Forms/UrlSchemesForm.cs         |    2 +-
 39 files changed, 45 insertions(+), 45 deletions(-)

diff --git a/KeePass/Forms/AboutForm.cs b/KeePass/Forms/AboutForm.cs
index 3e0ad1d..da4f0a4 100644
--- a/KeePass/Forms/AboutForm.cs
+++ b/KeePass/Forms/AboutForm.cs
@@ -56,12 +56,12 @@ namespace KeePass.Forms
 			string strTitle = PwDefs.ProductName;
 			string strDesc = KPRes.Version + " " + PwDefs.VersionString;
 
-			Icon icoNew = new Icon(Properties.Resources.KeePass, 48, 48);
+			Icon icoNew = new Icon(new Icon("/usr/share/keepass/KeePass.ico"), 48, 48);
 
 			m_bannerImage.Image = BannerFactory.CreateBanner(m_bannerImage.Width,
 				m_bannerImage.Height, BannerStyle.Default,
 				icoNew.ToBitmap(), strTitle, strDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			m_lvComponents.Columns.Add(KPRes.Components, 100, HorizontalAlignment.Left);
 			m_lvComponents.Columns.Add(KPRes.Version, 100, HorizontalAlignment.Left);
diff --git a/KeePass/Forms/CharPickerForm.cs b/KeePass/Forms/CharPickerForm.cs
index 870a252..00f4fdb 100644
--- a/KeePass/Forms/CharPickerForm.cs
+++ b/KeePass/Forms/CharPickerForm.cs
@@ -105,7 +105,7 @@ namespace KeePass.Forms
 
 			m_fontChars = new Font("Tahoma", 8.25f, FontStyle.Bold);
 
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			this.Text = KPRes.PickCharacters + " - " + PwDefs.ShortProductName;
 
 			m_secWord.Attach(m_tbSelected, OnSelectedTextChangedEx, true);
diff --git a/KeePass/Forms/ColumnsForm.cs b/KeePass/Forms/ColumnsForm.cs
index 3226713..b159327 100644
--- a/KeePass/Forms/ColumnsForm.cs
+++ b/KeePass/Forms/ColumnsForm.cs
@@ -56,7 +56,7 @@ namespace KeePass.Forms
 				m_bannerImage.Height, BannerStyle.Default,
 				Properties.Resources.B48x48_View_Detailed,
 				KPRes.ConfigureColumns, KPRes.ConfigureColumnsDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			this.Text = KPRes.ConfigureColumns;
 
 			int nWidth = (m_lvColumns.ClientRectangle.Width -
diff --git a/KeePass/Forms/DataEditorForm.cs b/KeePass/Forms/DataEditorForm.cs
index 7b2c867..38b2523 100644
--- a/KeePass/Forms/DataEditorForm.cs
+++ b/KeePass/Forms/DataEditorForm.cs
@@ -98,7 +98,7 @@ namespace KeePass.Forms
 
 			GlobalWindowManager.AddWindow(this);
 
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			this.DoubleBuffered = true;
 
 			m_bdc = BinaryDataClassifier.Classify(m_strDataDesc, m_pbData);
diff --git a/KeePass/Forms/DataViewerForm.cs b/KeePass/Forms/DataViewerForm.cs
index 35995c6..6431f1c 100644
--- a/KeePass/Forms/DataViewerForm.cs
+++ b/KeePass/Forms/DataViewerForm.cs
@@ -71,7 +71,7 @@ namespace KeePass.Forms
 
 			GlobalWindowManager.AddWindow(this);
 
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			m_tslViewer.Text = KPRes.ShowIn + ":";
 			m_tslEncoding.Text = KPRes.Encoding + ":";
diff --git a/KeePass/Forms/DatabaseOperationsForm.cs b/KeePass/Forms/DatabaseOperationsForm.cs
index e8dbc8b..1988645 100644
--- a/KeePass/Forms/DatabaseOperationsForm.cs
+++ b/KeePass/Forms/DatabaseOperationsForm.cs
@@ -62,7 +62,7 @@ namespace KeePass.Forms
 				m_bannerImage.Height, BannerStyle.Default,
 				Properties.Resources.B48x48_Package_Settings, KPRes.DatabaseMaintenance,
 				KPRes.DatabaseMaintenanceDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			this.Text = KPRes.DatabaseMaintenance;
 
 			m_numHistoryDays.Value = m_pwDatabase.MaintenanceHistoryDays;
diff --git a/KeePass/Forms/DatabaseSettingsForm.cs b/KeePass/Forms/DatabaseSettingsForm.cs
index 4589b57..eb0b111 100644
--- a/KeePass/Forms/DatabaseSettingsForm.cs
+++ b/KeePass/Forms/DatabaseSettingsForm.cs
@@ -78,7 +78,7 @@ namespace KeePass.Forms
 				m_bannerImage.Height, BannerStyle.Default,
 				Properties.Resources.B48x48_Ark, KPRes.DatabaseSettings,
 				KPRes.DatabaseSettingsDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			m_bInitializing = true;
 
diff --git a/KeePass/Forms/EcasTriggerForm.cs b/KeePass/Forms/EcasTriggerForm.cs
index c42b9d1..689b78c 100644
--- a/KeePass/Forms/EcasTriggerForm.cs
+++ b/KeePass/Forms/EcasTriggerForm.cs
@@ -68,7 +68,7 @@ namespace KeePass.Forms
 				m_bannerImage.Height, BannerStyle.Default,
 				Properties.Resources.B48x48_Run, strTitle, strDesc);
 			this.Text = strTitle;
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			m_lvEvents.SmallImageList = m_ilIcons;
 			m_lvConditions.SmallImageList = m_ilIcons;
diff --git a/KeePass/Forms/EcasTriggersForm.cs b/KeePass/Forms/EcasTriggersForm.cs
index 3c27e5d..4e97d60 100644
--- a/KeePass/Forms/EcasTriggersForm.cs
+++ b/KeePass/Forms/EcasTriggersForm.cs
@@ -83,7 +83,7 @@ namespace KeePass.Forms
 				Properties.Resources.B48x48_Make_KDevelop, KPRes.Triggers,
 				KPRes.TriggersDesc);
 			this.Text = KPRes.Triggers;
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			int nWidth = (m_lvTriggers.ClientSize.Width - UIUtil.GetVScrollBarWidth() - 1);
 			m_lvTriggers.Columns.Add(KPRes.Triggers, nWidth);
diff --git a/KeePass/Forms/EditAutoTypeItemForm.cs b/KeePass/Forms/EditAutoTypeItemForm.cs
index 7d4b6f9..8abd7b4 100644
--- a/KeePass/Forms/EditAutoTypeItemForm.cs
+++ b/KeePass/Forms/EditAutoTypeItemForm.cs
@@ -119,7 +119,7 @@ namespace KeePass.Forms
 					KPRes.ConfigureKeystrokeSeqDesc);
 			}
 
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			UIUtil.EnableAutoCompletion(m_cmbWindow, false);
 
diff --git a/KeePass/Forms/EditStringForm.cs b/KeePass/Forms/EditStringForm.cs
index 1492fe0..598d0d3 100644
--- a/KeePass/Forms/EditStringForm.cs
+++ b/KeePass/Forms/EditStringForm.cs
@@ -95,7 +95,7 @@ namespace KeePass.Forms
 			m_bannerImage.Image = BannerFactory.CreateBanner(m_bannerImage.Width,
 				m_bannerImage.Height, BannerStyle.Default,
 				Properties.Resources.B48x48_Font, strTitle, strDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			m_clrNormalBackground = m_cmbStringName.BackColor;
 
diff --git a/KeePass/Forms/EntropyForm.cs b/KeePass/Forms/EntropyForm.cs
index 0deb64a..fd90733 100644
--- a/KeePass/Forms/EntropyForm.cs
+++ b/KeePass/Forms/EntropyForm.cs
@@ -72,7 +72,7 @@ namespace KeePass.Forms
 				m_bannerImage.Height, BannerStyle.Default,
 				Properties.Resources.B48x48_Binary, KPRes.EntropyTitle,
 				KPRes.EntropyDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			this.Text = KPRes.EntropyTitle;
 
 			UpdateUIState();
diff --git a/KeePass/Forms/EntryListForm.cs b/KeePass/Forms/EntryListForm.cs
index a2a00ef..babf857 100644
--- a/KeePass/Forms/EntryListForm.cs
+++ b/KeePass/Forms/EntryListForm.cs
@@ -88,7 +88,7 @@ namespace KeePass.Forms
 				m_imgIcon, m_strTitle, m_strDescShort);
 			m_lblText.Text = m_strDescLong;
 			this.Text = m_strTitle;
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			if(m_ilIcons != null) m_lvEntries.SmallImageList = m_ilIcons;
 			else m_ilIcons = new ImageList();
diff --git a/KeePass/Forms/EntryReportForm.cs b/KeePass/Forms/EntryReportForm.cs
index 469b413..c805245 100644
--- a/KeePass/Forms/EntryReportForm.cs
+++ b/KeePass/Forms/EntryReportForm.cs
@@ -51,7 +51,7 @@ namespace KeePass.Forms
 
 		private void OnFormLoad(object sender, EventArgs e)
 		{
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			if(!string.IsNullOrEmpty(m_strTitle)) this.Text = m_strTitle;
 			else m_strTitle = PwDefs.ShortProductName;
diff --git a/KeePass/Forms/ExchangeDataForm.cs b/KeePass/Forms/ExchangeDataForm.cs
index eb665aa..c87d1e9 100644
--- a/KeePass/Forms/ExchangeDataForm.cs
+++ b/KeePass/Forms/ExchangeDataForm.cs
@@ -98,7 +98,7 @@ namespace KeePass.Forms
 				Properties.Resources.B48x48_Folder_Download);
 			m_bannerImage.Image = BannerFactory.CreateBanner(m_bannerImage.Width,
 				m_bannerImage.Height, BannerStyle.Default, bmpBanner, strWndTitle, strWndDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			this.Text = strWndTitle;
 
diff --git a/KeePass/Forms/FileBrowserForm.cs b/KeePass/Forms/FileBrowserForm.cs
index a75cf76..9b0d31c 100644
--- a/KeePass/Forms/FileBrowserForm.cs
+++ b/KeePass/Forms/FileBrowserForm.cs
@@ -93,7 +93,7 @@ namespace KeePass.Forms
 		{
 			GlobalWindowManager.AddWindow(this);
 
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			this.Text = m_strTitle;
 
 			m_nIconDim = m_tvFolders.ItemHeight;
diff --git a/KeePass/Forms/GroupForm.cs b/KeePass/Forms/GroupForm.cs
index 73473b0..129b155 100644
--- a/KeePass/Forms/GroupForm.cs
+++ b/KeePass/Forms/GroupForm.cs
@@ -67,7 +67,7 @@ namespace KeePass.Forms
 				m_bannerImage.Height, BannerStyle.Default,
 				Properties.Resources.B48x48_Folder_Txt, KPRes.EditGroup,
 				KPRes.EditGroupDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			UIUtil.SetButtonImage(m_btnAutoTypeEdit,
 				Properties.Resources.B16x16_Wizard, true);
diff --git a/KeePass/Forms/HelpSourceForm.cs b/KeePass/Forms/HelpSourceForm.cs
index f49d81a..c9c97e6 100644
--- a/KeePass/Forms/HelpSourceForm.cs
+++ b/KeePass/Forms/HelpSourceForm.cs
@@ -48,7 +48,7 @@ namespace KeePass.Forms
 				m_bannerImage.Height, BannerStyle.Default,
 				Properties.Resources.B48x48_Folder_Download, KPRes.HelpSourceSelection,
 				KPRes.HelpSourceSelectionDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			this.Text = KPRes.HelpSourceSelection;
 
 			FontUtil.AssignDefaultBold(m_radioLocal);
diff --git a/KeePass/Forms/IOConnectionForm.cs b/KeePass/Forms/IOConnectionForm.cs
index 16caca0..17e0668 100644
--- a/KeePass/Forms/IOConnectionForm.cs
+++ b/KeePass/Forms/IOConnectionForm.cs
@@ -77,7 +77,7 @@ namespace KeePass.Forms
 				m_bannerImage.Height, BannerStyle.Default,
 				KeePass.Properties.Resources.B48x48_WWW, strTitle,
 				strDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			this.Text = strTitle;
 
 			FontUtil.AssignDefaultBold(m_lblUrl);
diff --git a/KeePass/Forms/IconPickerForm.cs b/KeePass/Forms/IconPickerForm.cs
index 70bdafc..ec3596d 100644
--- a/KeePass/Forms/IconPickerForm.cs
+++ b/KeePass/Forms/IconPickerForm.cs
@@ -79,7 +79,7 @@ namespace KeePass.Forms
 
 			GlobalWindowManager.AddWindow(this);
 
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			FontUtil.AssignDefaultBold(m_radioStandard);
 			FontUtil.AssignDefaultBold(m_radioCustom);
diff --git a/KeePass/Forms/ImportCsvForm.cs b/KeePass/Forms/ImportCsvForm.cs
index 766c60f..51e8aa9 100644
--- a/KeePass/Forms/ImportCsvForm.cs
+++ b/KeePass/Forms/ImportCsvForm.cs
@@ -69,7 +69,7 @@ namespace KeePass.Forms
 
 			GlobalWindowManager.AddWindow(this);
 
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			m_bBlockChangedEvent = true;
 
diff --git a/KeePass/Forms/ImportMethodForm.cs b/KeePass/Forms/ImportMethodForm.cs
index 0c44925..f190bea 100644
--- a/KeePass/Forms/ImportMethodForm.cs
+++ b/KeePass/Forms/ImportMethodForm.cs
@@ -57,7 +57,7 @@ namespace KeePass.Forms
 				m_bannerImage.Height, BannerStyle.Default,
 				Properties.Resources.B48x48_Folder_Download, KPRes.ImportBehavior,
 				KPRes.ImportBehaviorDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			this.Text = KPRes.ImportBehavior;
 
diff --git a/KeePass/Forms/InternalBrowserForm.cs b/KeePass/Forms/InternalBrowserForm.cs
index f8fc293..428a4cb 100644
--- a/KeePass/Forms/InternalBrowserForm.cs
+++ b/KeePass/Forms/InternalBrowserForm.cs
@@ -53,7 +53,7 @@ namespace KeePass.Forms
 		{
 			GlobalWindowManager.AddWindow(this);
 
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			if(m_strInitialUrl.Length > 0)
 				m_webBrowser.Navigate(m_strInitialUrl);
diff --git a/KeePass/Forms/KeyCreationForm.cs b/KeePass/Forms/KeyCreationForm.cs
index 21982b9..02513c6 100644
--- a/KeePass/Forms/KeyCreationForm.cs
+++ b/KeePass/Forms/KeyCreationForm.cs
@@ -80,7 +80,7 @@ namespace KeePass.Forms
 				m_bannerImage.Height, BannerStyle.Default,
 				Properties.Resources.B48x48_KGPG_Sign, KPRes.CreateMasterKey,
 				m_ioInfo.GetDisplayName());
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			this.Text = KPRes.CreateMasterKey;
 
 			FontUtil.AssignDefaultBold(m_cbPassword);
diff --git a/KeePass/Forms/KeyPromptForm.cs b/KeePass/Forms/KeyPromptForm.cs
index 160de6f..0463261 100644
--- a/KeePass/Forms/KeyPromptForm.cs
+++ b/KeePass/Forms/KeyPromptForm.cs
@@ -120,7 +120,7 @@ namespace KeePass.Forms
 			m_bannerImage.Image = BannerFactory.CreateBanner(m_bannerImage.Width,
 				m_bannerImage.Height, BannerStyle.Default,
 				Properties.Resources.B48x48_KGPG_Key2, strBannerTitle, strBannerDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			FontUtil.AssignDefaultBold(m_cbPassword);
 			FontUtil.AssignDefaultBold(m_cbKeyFile);
diff --git a/KeePass/Forms/LanguageForm.cs b/KeePass/Forms/LanguageForm.cs
index ba302dc..8478a23 100644
--- a/KeePass/Forms/LanguageForm.cs
+++ b/KeePass/Forms/LanguageForm.cs
@@ -54,7 +54,7 @@ namespace KeePass.Forms
 				m_bannerImage.Height, BannerStyle.Default,
 				Properties.Resources.B48x48_Keyboard_Layout,
 				KPRes.SelectLanguage, KPRes.SelectLanguageDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			this.Text = KPRes.SelectLanguage;
 
 			int nWidth = m_lvLanguages.ClientRectangle.Width / 4;
diff --git a/KeePass/Forms/MainForm.cs b/KeePass/Forms/MainForm.cs
index 7e05058..441f06a 100644
--- a/KeePass/Forms/MainForm.cs
+++ b/KeePass/Forms/MainForm.cs
@@ -118,7 +118,7 @@ namespace KeePass.Forms
 			m_strNeverExpiresText = KPRes.NeverExpires;
 
 			this.Text = PwDefs.ProductName;
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			m_imgFileSaveEnabled = Properties.Resources.B16x16_FileSave;
 			m_imgFileSaveDisabled = Properties.Resources.B16x16_FileSave_Disabled;
 			// m_imgFileSaveAllEnabled = Properties.Resources.B16x16_File_SaveAll;
diff --git a/KeePass/Forms/MainForm_Functions.cs b/KeePass/Forms/MainForm_Functions.cs
index d8c84c7..708f815 100644
--- a/KeePass/Forms/MainForm_Functions.cs
+++ b/KeePass/Forms/MainForm_Functions.cs
@@ -374,12 +374,12 @@ namespace KeePass.Forms
 					strNtfPre.Length);
 
 				Icon icoDisposable, icoAssignable;
-				CreateColorizedIcon(Properties.Resources.QuadLocked, false,
+				CreateColorizedIcon(new Icon("/usr/share/keepass/QuadLocked.ico"), false,
 					ref m_kvpIcoTrayLocked, out icoAssignable, out icoDisposable);
 				m_ntfTray.Icon = icoAssignable;
 				if(icoDisposable != null) icoDisposable.Dispose();
 
-				TaskbarList.SetOverlayIcon(this, Properties.Resources.LockOverlay,
+				TaskbarList.SetOverlayIcon(this, new Icon("/usr/share/keepass/LockOverlay.ico"),
 					KPRes.Locked);
 				NativeMethods.EnableWindowPeekPreview(this.Handle, false);
 			}
@@ -389,7 +389,7 @@ namespace KeePass.Forms
 				strNtfText = strWindowText;
 
 				Icon icoDisposable, icoAssignable;
-				CreateColorizedIcon(Properties.Resources.QuadNormal, false,
+				CreateColorizedIcon(new Icon("/usr/share/keepass/QuadNormal.ico"), false,
 					ref m_kvpIcoTrayNormal, out icoAssignable, out icoDisposable);
 				m_ntfTray.Icon = icoAssignable;
 				if(icoDisposable != null) icoDisposable.Dispose();
@@ -419,7 +419,7 @@ namespace KeePass.Forms
 					m_docMgr.ActiveDatabase.IOConnectionInfo.Path, 63 - strNtfPre.Length);
 
 				Icon icoDisposable, icoAssignable;
-				CreateColorizedIcon(Properties.Resources.QuadNormal, false,
+				CreateColorizedIcon(new Icon("/usr/share/keepass/QuadNormal.ico"), false,
 					ref m_kvpIcoTrayNormal, out icoAssignable, out icoDisposable);
 				m_ntfTray.Icon = icoAssignable;
 				if(icoDisposable != null) icoDisposable.Dispose();
@@ -434,7 +434,7 @@ namespace KeePass.Forms
 			m_ntfTray.Text = StrUtil.CompactString3Dots(strNtfText, 63);
 
 			Icon icoToDispose, icoToAssign;
-			if(CreateColorizedIcon(Properties.Resources.KeePass, true,
+			if(CreateColorizedIcon(new Icon("/usr/share/keepass/KeePass.ico"), true,
 				ref m_kvpIcoMain, out icoToAssign, out icoToDispose))
 				this.Icon = icoToAssign;
 			if(icoToDispose != null) icoToDispose.Dispose();
@@ -3245,7 +3245,7 @@ namespace KeePass.Forms
 					// wasn't fully constructed at that time yet)
 					if(IsFileLocked(null))
 						TaskbarList.SetOverlayIcon(this,
-							Properties.Resources.LockOverlay, KPRes.Locked);
+							new Icon("/usr/share/keepass/LockOverlay.ico"), KPRes.Locked);
 				}
 
 				if(Program.Config.MainWindow.MinimizeToTray) MinimizeToTray(true);
diff --git a/KeePass/Forms/OptionsForm.cs b/KeePass/Forms/OptionsForm.cs
index d23e3e1..68ef6d4 100644
--- a/KeePass/Forms/OptionsForm.cs
+++ b/KeePass/Forms/OptionsForm.cs
@@ -104,7 +104,7 @@ namespace KeePass.Forms
 		{
 			GlobalWindowManager.AddWindow(this);
 
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			Debug.Assert(m_ilIcons != null);
 			if(m_ilIcons != null)
diff --git a/KeePass/Forms/PluginsForm.cs b/KeePass/Forms/PluginsForm.cs
index cb2db6e..96c4d43 100644
--- a/KeePass/Forms/PluginsForm.cs
+++ b/KeePass/Forms/PluginsForm.cs
@@ -65,7 +65,7 @@ namespace KeePass.Forms
 				m_bannerImage.Height, BannerStyle.Default,
 				Properties.Resources.B48x48_BlockDevice, KPRes.Plugins,
 				KPRes.PluginsDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			m_cbCacheDeleteOld.Checked = Program.Config.Application.Start.PluginCacheDeleteOld;
 
diff --git a/KeePass/Forms/PrintForm.cs b/KeePass/Forms/PrintForm.cs
index 0f2e2a6..5d0e60c 100644
--- a/KeePass/Forms/PrintForm.cs
+++ b/KeePass/Forms/PrintForm.cs
@@ -96,7 +96,7 @@ namespace KeePass.Forms
 
 			GlobalWindowManager.AddWindow(this);
 
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			CreateDialogBanner();
 
 			UIUtil.SetButtonImage(m_btnConfigPrinter,
diff --git a/KeePass/Forms/PwEntryForm.cs b/KeePass/Forms/PwEntryForm.cs
index a34d423..c8d2534 100644
--- a/KeePass/Forms/PwEntryForm.cs
+++ b/KeePass/Forms/PwEntryForm.cs
@@ -547,7 +547,7 @@ namespace KeePass.Forms
 				BannerStyle.Default,
 				KeePass.Properties.Resources.B48x48_KGPG_Sign,
 				strTitle, strDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			this.Text = strTitle;
 
 			UIUtil.SetButtonImage(m_btnTools,
diff --git a/KeePass/Forms/PwGeneratorForm.cs b/KeePass/Forms/PwGeneratorForm.cs
index b69a014..5142164 100644
--- a/KeePass/Forms/PwGeneratorForm.cs
+++ b/KeePass/Forms/PwGeneratorForm.cs
@@ -91,7 +91,7 @@ namespace KeePass.Forms
 				m_bannerImage.Height, BannerStyle.Default,
 				Properties.Resources.B48x48_KGPG_Gen, KPRes.PasswordOptions,
 				KPRes.PasswordOptionsDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			UIUtil.SetButtonImage(m_btnProfileAdd,
 				Properties.Resources.B16x16_FileSaveAs, false);
diff --git a/KeePass/Forms/SearchForm.cs b/KeePass/Forms/SearchForm.cs
index ecaf06d..5c30687 100644
--- a/KeePass/Forms/SearchForm.cs
+++ b/KeePass/Forms/SearchForm.cs
@@ -91,7 +91,7 @@ namespace KeePass.Forms
 				m_bannerImage.Height, BannerStyle.Default,
 				Properties.Resources.B48x48_XMag, strTitle,
 				KPRes.SearchDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			SearchParameters sp = Program.Config.Defaults.SearchParameters;
 			m_cbTitle.Checked = sp.SearchInTitles;
diff --git a/KeePass/Forms/SingleLineEditForm.cs b/KeePass/Forms/SingleLineEditForm.cs
index e0e4a95..ca4e84a 100644
--- a/KeePass/Forms/SingleLineEditForm.cs
+++ b/KeePass/Forms/SingleLineEditForm.cs
@@ -74,7 +74,7 @@ namespace KeePass.Forms
 			m_bannerImage.Image = BannerFactory.CreateBanner(m_bannerImage.Width,
 				m_bannerImage.Height, BannerStyle.Default,
 				m_imgIcon, m_strTitle, m_strDesc);
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 
 			this.Text = m_strTitle;
 			m_lblLongDesc.Text = m_strLongDesc;
diff --git a/KeePass/Forms/StatusLoggerForm.cs b/KeePass/Forms/StatusLoggerForm.cs
index 86e9e95..ebb9cc7 100644
--- a/KeePass/Forms/StatusLoggerForm.cs
+++ b/KeePass/Forms/StatusLoggerForm.cs
@@ -124,7 +124,7 @@ namespace KeePass.Forms
 		{
 			GlobalWindowManager.AddWindow(this);
 
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			this.Text = PwDefs.ShortProductName;
 			
 			m_pbProgress.Minimum = 0;
diff --git a/KeePass/Forms/TanWizardForm.cs b/KeePass/Forms/TanWizardForm.cs
index 4740442..e61fef9 100644
--- a/KeePass/Forms/TanWizardForm.cs
+++ b/KeePass/Forms/TanWizardForm.cs
@@ -62,7 +62,7 @@ namespace KeePass.Forms
 				KeePass.Properties.Resources.B48x48_Wizard, KPRes.TanWizard,
 				KPRes.TanWizardDesc);
 			
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			this.Text = KPRes.TanWizard;
 
 			if((m_pgStorage.Name != null) && (m_pgStorage.Name.Length > 0))
diff --git a/KeePass/Forms/UrlSchemeForm.cs b/KeePass/Forms/UrlSchemeForm.cs
index 4ff62a8..6b88f6d 100644
--- a/KeePass/Forms/UrlSchemeForm.cs
+++ b/KeePass/Forms/UrlSchemeForm.cs
@@ -51,7 +51,7 @@ namespace KeePass.Forms
 
 			GlobalWindowManager.AddWindow(this);
 
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			this.Text = KPRes.UrlSchemeOverride;
 
 			m_tbScheme.Text = m_ovr.Scheme;
diff --git a/KeePass/Forms/UrlSchemesForm.cs b/KeePass/Forms/UrlSchemesForm.cs
index eadc777..c605501 100644
--- a/KeePass/Forms/UrlSchemesForm.cs
+++ b/KeePass/Forms/UrlSchemesForm.cs
@@ -54,7 +54,7 @@ namespace KeePass.Forms
 
 			GlobalWindowManager.AddWindow(this);
 
-			this.Icon = Properties.Resources.KeePass;
+			this.Icon = new Icon("/usr/share/keepass/KeePass.ico");
 			this.Text = KPRes.UrlSchemeOverrides;
 
 			int nWidth = (m_lvOverrides.ClientSize.Width - UIUtil.GetVScrollBarWidth()) / 4;
-- 
